### ISSUE TITLE
React v2.1.1 positions ListboxOptions without anchor attribute set

### ISSUE THREAD

**Issue by RobinMalfait** [tailwindlabs/headlessui#3336]
### Discussed in https://github.com/tailwindlabs/headlessui/discussions/3335

<div type='discussions-op-text'>

<sup>Originally posted by **catyperry** June 27, 2024</sup>
In version 2.1.0 when I wanted to position my ListboxOptions myself I could just to it like this:

```react
<Listbox value={selected} onChange={(value) => setSelected(value)}>
  <div className="relative">
    <ListboxButton>
      <span>{selected.label}</span>
    </ListboxButton>
    <ListboxOptions className="absolute z-10 mt-1 max-h-60 w-full overflow-auto">
      {options.map((option) => (
        <ListboxOption key={option.id} value={option}>
              {option.label}
        </ListboxOption>
      ))}
    </ListboxOptions>
  </div>
</Listbox>
```

no issue here. But in version 2.1.1 a style prop with extra styles overwrites my ListboxOptions classes:
<img width="489" alt="Bildschirmfoto 2024-06-27 um 12 05 12" src="https://github.com/tailwindlabs/headlessui/assets/16886835/d26fb0c9-81e4-4cda-be7a-914c70aba5fd">

The `left` `top` props cause my ListboxOptions to be rendered on top of my ListboxButton instead of below it. 
I especially do this because I want to control the `max-height`of my ListboxOptions and its overflow (I was not able to get this to work in 2.1.1). And also because I like that this is more explicit imo, in contrast to working with css variables for `width` and positioning stuff.

Is this new behavior of setting inline styles regardless of whether the anchor prop is set or not intentional? If so i would have to use the anchor prop with the css variables. But this leaves me with the question of how to control the overflow and `max-height` of the ListboxOptions. I couldn't find a propper answer in the docs.

I very much appreciate every help 🙏</div>

**Comment by RobinMalfait** [comment]
This should be fixed by #3337, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.

**Comment by mohammadzolfali** [comment]
<Listbox value={selected} onChange={(value) => setSelected(value)}>
  <div className="relative">
    <ListboxButton>
      <span>{selected.label}</span>
    </ListboxButton>
    <ListboxOptions className="absolute z-10 mt-1 max-h-60 w-full overflow-auto">
      {options.map((option) => (
        <ListboxOption key={option.id} value={option}>
              {option.label}
        </ListboxOption>
      ))}
    </ListboxOptions>
  </div>
</Listbox>
